Jetcaster/app/src/main/java/com/example/jetcaster/ui/home/discover/Discover.kt

```diff
@@ -84,20 +81,21 @@ private val emptyTabIndicator: @Composable (List<TabPosition>) -> Unit = {}
 
 @Composable
 private fun PodcastCategoryTabs(
-    categories: List<Category>,
-    selectedCategory: Category,
+    filterableCategoriesModel: FilterableCategoriesModel,
     onCategorySelected: (Category) -> Unit,
     modifier: Modifier = Modifier
 ) {
-    val selectedIndex = categories.indexOfFirst { it == selectedCategory }
+    val selectedIndex = filterableCategoriesModel.categories.indexOfFirst {
+        filterableCategoriesModel.selectedCategory == it
+    }
```

Comment:
You can simplify this to standard `indexOf` 

```suggestion
    val selectedIndex = filterableCategoriesModel.categories.indexOf(filterableCategoriesModel.selectedCategory)
```

Or you could even encapsulate it into the `FilterableCategoriesModel` as a `val selectedCategoryIndex: Int = categories.indexOf(selectedCategory)`